internal/controller/dispatch_logic.go

```diff
@@ -78,12 +80,21 @@ func (r *AppWrapperReconciler) listAppWrappers(ctx context.Context) (map[int]Wei
 	}
 	requests := map[int]Weights{}        // total request per priority level
 	queue := []*mcadv1beta1.AppWrapper{} // queued appWrappers
+
+	appWrapperCount := map[phaseStepPriority]int{}
+
 	for _, appWrapper := range appWrappers.Items {
 		// get phase from cache if available as reconciler cache may be lagging
 		phase, step := r.getCachedPhase(&appWrapper)
+		priority := int(appWrapper.Spec.Priority)
+		key := phaseStepPriority{phase, step, priority}
+		if _, exists := appWrapperCount[key]; !exists {
+			appWrapperCount[key] = 0
```

Comment:
Is it OK to piggyback this loop and the `listAppWrappers()` method to do the counting for the new metric?

Reply:
It is probably OK once #64 is merged ensuring that metrics cannot age beyond `dispatchDelay` but we may want to do the metrics collection as a separate reconciliation cycle with its own trigger logic distinct from the `triggerDispatch` logic. For instance with the current approach, the failure of an AppWrapper may only be reflected in the metrics after the `dispatchDelay` timeout, which may or may not be reactive enough from a user's perspective.

Reply:
I see. Would you like the separate reconciliation cycle of metrics to be part of this PR or as a new PR?
Anyway, I think I would need more help or guidance as I'm not sure how to do it.